### PR TITLE
CSS Code Housekeeping

### PR DESCRIPTION
**Pull request by SychO9**
**Progresses #2600**

**Changes proposed in this pull request:**
General cleanup of certain bits of code, all safe and doesn't change anything.
* Removed dead code.
* Used CSS Variables in certain scopes to improve customizability.

**Reviewers should focus on:**
🤷🏼‍♂️ 

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.

### CODE REVIEW

**Comment by davwheat** [comment]
I'm definitely in favour of this direction. We should also remove all our vendor mixins (`.transition()`, `.box-shadow()`) as part of this PR.

**Comment by SychO9** [comment]
> I'm definitely in favour of this direction. We should also remove all our vendor mixins (`.transition()`, `.box-shadow()`) as part of this PR.

Yes as part of this PR and not others if you see what I mean :P

**Comment by SychO9** [comment]
Actually, removing the mixins would break any extensions using them 🤔

I'll just remove their usage for now.

**Review by davwheat — CHANGES_REQUESTED** [review]
We might want to remove the usages of translate3d. We can use normal translate instead.

This was a hack to force a separate compositing layer which was GPU optimised, but modern browsers do this automatically if transform is animated in any way in CSS.

We should also remove the unneeded `~""` surrounding some of the property values.

Couldn't be modify the mixing to only create the unprefixed version of each of the properties? That way extensions using them don't break, but they don't have the vendor prefixes either.

File: less/common/Modal.less [hunk 1]
```diff
@@ -37,11 +37,11 @@
...
   // When fading in the modal, animate it to slide down
   .Modal {
-    .scale(0.9);
-    .transition-transform(0.2s ease-out);
+    transform: scale(0.9);
+    transition: transform 0.2s ease-out;
   }
   &.in .Modal {
-    .scale(1);
+    transform: scale(0.9);
```
Comment (davwheat) [inline less/common/Modal.less:44]:
```suggestion
    transform: scale(1);
```

File: less/common/Modal.less [hunk 2]
```diff
@@ -126,7 +126,7 @@
   opacity: 0;
   pointer-events: none;
   border-radius: @border-radius;
-  .transition(opacity 0.2s);
+  transition: 0.2s opacity;
```
Comment (davwheat) [inline less/common/Modal.less:129]: For consistency

```suggestion
  transition: opacity 0.2s;
```

File: less/common/Navigation.less [hunk 3]
```diff
@@ -38,7 +38,7 @@
     }
   }
   .hasPane.panePinned .Navigation-pin .icon {
-    .rotate(0deg);
+    transform: rotate(0deg);
```
Comment (davwheat) [inline less/common/Navigation.less:41]:
```suggestion
    transform: rotate(0);
```

File: less/common/Search.less [hunk 4]
```diff
@@ -59,7 +59,7 @@
     width: 225px;
     padding-left: 32px;
     padding-right: 32px;
-    .transition(all 0.4s);
+    transition: all 0.4s;
```
Comment (davwheat) [inline less/common/Search.less:62]: We should probably avoid `all`. It's not your doing, but I wouldn't be opposed to you changing it ;)

https://www.joshwcomeau.com/animation/css-transitions/

File: less/forum/PostStream.less [hunk 5]
```diff
@@ -28,8 +28,8 @@
   100% {opacity: 0.5}
 }
 .LoadingPost {
-  .animation(blink 1s linear);
-  .animation-iteration-count(infinite);
+  animation: blink 1s linear;
```
Comment (davwheat) [inline less/forum/PostStream.less:31]: We can combine these two lines into one. Same goes for below.

File: less/forum/PostStream.less [hunk 6]
```diff
@@ -63,18 +63,6 @@
...
-//     position: absolute;
-//     left: -30px;
-//     top: -5px;
-//     bottom: -5px;
-//     width: 5px;
-//     border-radius: @border-radius;
-//     background: @fl-primary-color;
-//   }
-// }
 @-webkit-keyframes pulsate {
```
Comment (davwheat) [inline less/forum/PostStream.less:66]: We can remove the WebKit version of this.

File: less/forum/SignUpModal.less [hunk 7]
```diff
@@ -1,48 +0,0 @@
-.SignUpModal-welcome {
```
Comment (davwheat) [inline less/forum/SignUpModal.less:1]: Was this meant to be deleted?

**Review by davwheat — CHANGES_REQUESTED** [review]
Sorry about the multiple reviews. My fat fingers kept pressing the wrong button on mobile.

File: less/common/Dropdown.less [hunk 8]
```diff
@@ -176,12 +176,12 @@
...
     border-radius: 0;
-    .box-shadow(0 2px 6px @shadow-color);
+    box-shadow: 0 2px 6px @shadow-color;
     visibility: hidden;
     overflow: auto;
     -webkit-overflow-scrolling: touch;
-    .translate3d(0, 70vh, 0);
-    .transition-transform(~" 0.3s, visibility 0s 0.3s");
+    transform: translate3d(0, 70vh, 0);
+    transition: transform ~" 0.3s, visibility 0s 0.3s";
```
Comment (SychO9) [inline less/common/Dropdown.less:184]:
```suggestion
    transition: transform 0.3s, visibility 0s 0.3s;
```

File: less/common/Dropdown.less [hunk 9]
```diff
@@ -222,8 +222,8 @@
   .dropdown-backdrop {
     background: fade(@secondary-color, 90%);
     opacity: 0;
-    .transition(~"opacity 0.3s");
-    .translate3d(0, 0, 0);
+    transition: ~"opacity 0.3s";
```
Comment (SychO9) [inline less/common/Dropdown.less:225]:
```suggestion
    transition: opacity 0.3s;
```

File: less/common/FormControl.less [hunk 10]
```diff
@@ -9,7 +9,7 @@
   background-color: @control-bg;
   border: 2px solid transparent;
   border-radius: @border-radius;
-  .transition(~"border-color .15s, background .15s");
+  transition: ~"border-color .15s, background .15s";
```
Comment (SychO9) [inline less/common/FormControl.less:12]:
```suggestion
  transition: border-color .15s, background .15s;
```

File: less/common/Modal.less [hunk 11]
```diff
@@ -16,7 +16,7 @@
   z-index: @zindex-modal-background;
   background-color: @overlay-bg;
   opacity: 0;
-  .transition(0.2s opacity);
+  transition: 0.2s opacity;
```
Comment (SychO9) [inline less/common/Modal.less:19]:
```suggestion
  transition: opacity 0.2s;
```

File: less/common/Navigation.less [hunk 12]
```diff
@@ -17,10 +17,10 @@
   opacity: 0;
   margin-left: -5px !important;
   border-radius: 0 @border-radius @border-radius 0;
-  .transition(~"opacity 0.2s, margin-left 0.2s");
+  transition: ~"opacity 0.2s, margin-left 0.2s";
```
Comment (SychO9) [inline less/common/Navigation.less:20]:
```suggestion
  transition: opacity 0.2s, margin-left 0.2s;
```

File: less/common/Tooltip.less [hunk 13]
```diff
@@ -10,7 +10,7 @@
   font-weight: normal;
   line-height: 1.4;
   opacity: 0;
-  .transition(0.15s opacity linear);
+  transition: 0.15s opacity linear;
```
Comment (SychO9) [inline less/common/Tooltip.less:13]:
```suggestion
  transition: opacity 0.15s linear;
```

File: less/common/mixins/header-background.less [hunk 14]
```diff
@@ -6,7 +6,7 @@
   right: 0;
   z-index: @zindex-header;
   border-bottom: 1px solid @control-bg;
-  .transition(~"box-shadow 0.2s, -webkit-transform 0.2s");
+  transition: ~"box-shadow 0.2s, -webkit-transform 0.2s";
```
Comment (SychO9) [inline less/common/mixins/header-background.less:9]: Not sure what the `-webkit-transform` here is for, there isn't even a transform property 🤔 but either way we can change that to `transform`.

```suggestion
  transition: box-shadow 0.2s, transform 0.2s;
```

File: less/forum/Composer.less [hunk 15]
```diff
@@ -221,20 +221,20 @@
...
-    .transition(left 0.2s);
+    transition: left 0.2s;
   }
   .Composer {
     border-radius: @border-radius @border-radius 0 0;
     background: fade(@body-bg, 95%);
     position: relative;
     height: 300px;
-    .transition(~"background 0.2s, box-shadow 0.2s");
+    transition: ~"background 0.2s, box-shadow 0.2s";
```
Comment (SychO9) [inline less/forum/Composer.less:231]:
```suggestion
    transition: background 0.2s, box-shadow 0.2s;
```

File: less/forum/Composer.less [hunk 16]
```diff
@@ -221,20 +221,20 @@
...
 
     &.active, &.fullScreen {
       background: @body-bg;
     }
     &.active:not(.fullScreen) {
-      .box-shadow(~"0 0 0 2px @{primary-color}, 0 2px 6px @{shadow-color}");
+      box-shadow: ~"0 0 0 2px @{primary-color}, 0 2px 6px @{shadow-color}";
```
Comment (SychO9) [inline less/forum/Composer.less:237]:
```suggestion
      box-shadow: 0 0 0 2px @primary-color, 0 2px 6px @shadow-color;
```

File: less/forum/Post.less [hunk 17]
```diff
@@ -65,7 +65,7 @@
     top: -10px;
     left: -100px;
     z-index: @zindex-dropdown;
-    .transition(~"opacity 0.2s, transform 0.2s");
+    transition: ~"opacity 0.2s, transform 0.2s";
```
Comment (SychO9) [inline less/forum/Post.less:68]:
```suggestion
    transition: opacity 0.2s, transform 0.2s;
```

File: less/forum/Slidable.less [hunk 18]
```diff
@@ -40,14 +40,14 @@
       left: unset;
     }
     .Slidable-content {
-      .transition(~"box-shadow 0.2s, border-radius 0.2s");
+      transition: ~"box-shadow 0.2s, border-radius 0.2s";
```
Comment (SychO9) [inline less/forum/Slidable.less:43]:
```suggestion
      transition: box-shadow 0.2s, border-radius 0.2s;
```

**Comment by SychO9** [comment]
> We might want to remove the usages of translate3d. We can use normal translate instead.
>
> This was a hack to force a separate compositing layer which was GPU optimised, but modern browsers do this automatically if transform is animated in any way in CSS.

I couldn't find anything online about this to make sure it's no longer necessary :/

> Couldn't be modify the mixing to only create the unprefixed version of each of the properties? That way extensions using them don't break, but they don't have the vendor prefixes either.

The deprecated mixins here are all vendor-less though, you removed them in a previous PR.

**Comment by SychO9** [comment]
This should be enough for now.

**Review by davwheat — APPROVED** [review]

**Review by askvortsov1 — APPROVED** [review]

**Pull request merged**